app/src/org/commcare/android/framework/SessionAwareCommCareActivity.java

```diff
@@ -1,11 +1,30 @@
 package org.commcare.android.framework;
 
+import android.os.Bundle;
+
+import org.commcare.android.util.SessionUnavailableException;
+
 /**
  * Manage redirection to login screen when session expiration occurs.
  *
  * @author Phillip Mates (pmates@dimagi.com)
  */
 public abstract class SessionAwareCommCareActivity<R> extends CommCareActivity<R> {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        try {
+            onCreateAware(savedInstanceState);
+        } catch (SessionUnavailableException e) {
+            SessionActivityRegistration.redirectToLogin(this);
+        }
```

Comment:
I think we should try and find an alternative to exposing `onCreateAware` and wrapping its call in a try/catch.
When possible it is good to avoid wrapping large pieces of code in try/catch statements because doing so lets us be lazy about when things fail.

I'll poke around and see if I can come up with any ideas.